core/modules/widgets/navigator.js

```diff
@@ -60,6 +60,10 @@ NavigatorWidget.prototype.execute = function() {
 	// Get our parameters
 	this.storyTitle = this.getAttribute("story");
 	this.historyTitle = this.getAttribute("history");
+	this.openLinkFromInsideRiver = this.getAttribute("openLinkFromInsideRiver","top");
+	this.openLinkFromOutsideRiver = this.getAttribute("openLinkFromOutsideRiver","top");
+	this.singleTiddlerMode = this.getAttribute("singleTiddlerMode","no") === "yes";
+	this.relinkOnRename = this.getAttribute("relinkOnRename","no").toLowerCase().trim() === "yes";
 	this.setVariable("tv-story-list",this.storyTitle);
```

Comment:
this is all super ugly!  IMO it shouldnt be that way

Reply:
This is consistent with how all the core widgets manage their attributes.

Reply:
Yes, but that's not the point I wanted to make. ... The navigator widget shouldn't be responsible to handle views. It's the story and storyviews in combination, which need to handle the stuff. 

So If an event wants to pass any parameter through, it need to be dynamic and not hardcoded like this one. 

We do have the `paramObject` which can hold any additional parameters the users and plugin authors may want to send to the navigator event handler. 

I want the stuff to be dynamic and there shouldn't be a core PR necessare if new parameters like this one are needed by a new function. 

It's OK if you make it. It gets merged, but if anyone else wants new params the chance is way to low. ... So I do want a solution, similar to the linkcatcher that I used for the new "story manager stuff" I'm creating. see: https://github.com/pmario/TiddlyWiki5/blob/dbf32fdef66196fb389ad55665b2f2c84991b65f/core/modules/widgets/linkcatcher.js#L75

linkcatcher has parameters that are OK, but can't be used for the new `tm-load-story` message. I needed different params. With the mechanism I propose it will be simple, to create new tm-xxx plugins without the need to modify linkcatcher or navigator. They just pass through the `paramObject`

also see: https://github.com/wikilabs/plugins/blob/8b85e7b8d66b0b4dd7c7f3fc213a22501e633f44/wikilabs/wltm-open-story/tiddlers/open-story.js#L42 where the paramObject defined above AND sent by a user can be used. 

Usage can be see at: https://github.com/wikilabs/editions/blob/eca970252a441c2612239453c01d406da7519d75/wltm-open-story/tiddlers/%24__core_ui_AdvancedSearch_Story_Tabs_template.tid#L8

and: https://github.com/wikilabs/editions/blob/eca970252a441c2612239453c01d406da7519d75/wltm-open-story/tiddlers/%24__core_ui_SideBar_Stories.tid#L20

I think, that's the mechanism we should use to add unknown future parameters to navigator and story related widgets.

Reply:
Hi @pmario I think you're proposing a mechanism whereby widgets that dispatch events can specify the `paramObject` hashmap by using unknown attributes. Why wouldn't one use the `action-sendmessage` widget in such situations?

I don't quite understand how the proposal would be applied in this case? Which message is involved?